readme.md

```diff
@@ -0,0 +1,182 @@
+
+By taking in both **2D RGB images** with **3D PointCloud** information, RVIP combines the high effieciency and accuracy of 2D computer vision architectures with the robust data points of depth information.
+
+This is done so in an effort to ensure higher reliability and reduced volatility of object localization results, needed for safer robot-human collaboration spaces.
+
+For a detailed documentation on RVIP, please execute the following commands:
+1. Clone this repository into the `src` folder of your catkin workspace.
+
+> ` $ git clone https://github.com/cardboardcode/rvip.git`
+
+2. Install **rosdoc_lite**.
+
+> `$ sudo apt install ros-melodic-rosdoc-lite`
+
+3. Go to the `rvip` subdirectory and run rosdoc_lite command
+
+> `$ cd rvip && rosdoc_lite .`
+
+4. View the documentation on your browser.
+
+> `$ firefox doc/html/doc/index.html `
+
+## How Does It Work?
+
+The flow of data using RVIP is illustrated in the following diagram.
+
+**Inputs**:
+1. 3D PointCloud (sensor_msgs::PointCloud2)
+2. 2D RGB Camera Image (sensor_msgs::Image)
+
+**Outputs**:
+1. Object Centroid Position, Orientation and Dimensions (visualisation_msgs::MarkerArray,tf::Transform)
+
+![](/rvip/img/rvip_architecture.jpg)
+<center>RVIP Architecture Diagram</center>
+
+### Stages
+**[1]** - **3D PointCloud** and rectified **2D RGB Camera Image** are received from a 3D stereo-vision camera such as the Kinect ONE V2.
+
+**[2]** - The rectified **2D RGB Camera Image** are given to an external ROS package that houses its own 2D Computer Vision Architecture. An example would be [darknet_ros](https://github.com/leggedrobotics/darknet_ros).
+
+**[3]** - The role of RVIP starts here by receiving the synchronized pair of 3D PointCloud and the RegionOfInterest (RoI).
+1. RVIP filters out valid RoI to avoid duplicates.
+2. RVIP extracts the PointCloud cluster based on the given RoI information by segmenting the part of cluster that falls within the bounding box defined by the RoI.
+3. RVIP extracts the lower larger plane of PointCloud cluster to determine the lower z-limit.
+4. RVIP extracts the object's top plane to deter the upper z-limit.
+
+**[4]** - RVIP estimates the length and width of the detected object using the object's top plane.
+RVIP estimates the height by subtracting the upper and lower z-limits.
+
+**[5]** - Based on the length, width and height of the object, a cuboid PointCloud is generated. A minimally 2-sided PointCloud is also extracted and used for the pose alignment later.
+
+**[6]** - RVIP executes the pose alignment with [SUPER-4PCS](https://github.com/STORM-IRIT/OpenGR) as well as [Iterative Closest Point (ICP)](http://pointclouds.org/documentation/tutorials/iterative_closest_point.php)
+
+**[7]** - RVIP finally outputs the object centroid position (float x,y,z), object orientation (Quaternion) and its dimensions.
+
+****
+
+### Requirements
+#### This section lists all third-party softwares used by **RVIP**.
```

Comment:
How is this related to the same (?) content in `demo.md`?

Reply:
The repeat of this section in both **demo.md** and **readme.md** serves as a convenience for users so they need not refer to both documents at the same time during **rvip** setup.

Reply:
I would really suggest to find a way to avoid the duplication between these files. Any change to the readme will have to be duplicated to all the other copies.

Perhaps to address the general challenge of novices vs more experienced users: perhaps the main documentation (ie: the readme) could start with a section where you mention the existence of (much) more detailed documentation. For novices, you recommend they follow this.

More experienced users could continue with the -- much less verbose -- "quick start guide" or whatever name you want to use.

+1 for thinking of less experienced users, but right now there is so much text I can't even quickly determine what I need to do to get things to build and run.

Reply:
Addressed in [this commit](https://github.com/ros-industrial/rvip/pull/1/commits/877a7ec8dddc672c7794b5310f302911899ca83e).